### PR TITLE
Feature/add circleci

### PR DESCRIPTION
**Pull request by m3ller**
* Add CI for maze
* Ignore "Add Apache License" commit as after rebasing with upstream, license was already taken care of

* Will test with py3.8 and windows in issue #1

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+  test-osx-2.7:
+    <<: *osx-tests-template
+    environment:
+      PYTHON: 2.7.15
+      HOMEBREW_NO_AUTO_UPDATE: 1
+      BASH_ENV: /Users/distiller/repo/.circleci/.bashrc_macOS
+
+
+  deploy:
```
Comment (randomir) [inline .circleci/config.yml]: We're not deploying demos.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      - test-3.7
+      - test-3.6
+      - test-3.5
+      - test-2.7
+      - test-osx-3.7
+      - test-osx-3.6
+      - test-osx-3.5
+      - test-osx-2.7
+
+  deploy:
```
Comment (randomir) [inline .circleci/config.yml]: No deploy.

**Review by randomir — APPROVED** [review]
Wanna try adding windows to CircleCI tests?

**Comment by m3ller** [comment]
Yes, @randomir, I will get windows and py 3.8 in another PR, #1 . I just want to do the easy thing and get the "old" version of the tests in first.

**Pull request merged**